src/frontends/common/include/openvino/frontend/extension/op.hpp

```diff
@@ -234,29 +345,71 @@ OpExtensionBase<BaseConversionType, OVOpType>::OpExtensionBase(const std::string
                              attr_names_map,
                              attr_values_map)) {}
 
+template <typename BaseConversionType, typename OVOpType>
+OpExtensionBase<BaseConversionType, OVOpType>::OpExtensionBase(const std::string& fw_type_name,
+                                                               const std::vector<std::string>& in_names_vec,
+                                                               const std::vector<std::string>& out_names_vec,
+                                                               const std::map<std::string, std::string>& attr_names_map,
+                                                               const std::map<std::string, ov::Any>& attr_values_map)
+    : BaseConversionType(fw_type_name,
+                         OpConversionFunctionNamed(
+                             []() {
+                                 return std::make_shared<OVOpType>();
+                             },
+                             in_names_vec,
+                             out_names_vec,
+                             attr_names_map,
+                             attr_values_map)) {}
+
 template <typename OVOpType = void>
 using OpExtension = ov::frontend::OpExtensionBase<ov::frontend::ConversionExtension, OVOpType>;
 
+#define FRONTEND_EXPAND(X)  X
+#define FRONTEND_CAT_(x, y) x##y
+#define FRONTEND_CAT(x, y)  FRONTEND_CAT_(x, y)
+// extract common attribute and values
+#define GEN_VAR_COMMON(...)                     \
+    auto params = make_spec_tuple(__VA_ARGS__); \
+    const auto& name = std::get<0>(params);     \
+    const auto& attr_mp = std::get<1>(params);  \
+    const auto& val_mp = std::get<2>(params);
+// extract paddle specific param + common values
+#define GEN_VAR_PADDLE(in_names, out_names, ...)             \
+    const std::vector<std::string> in_names_vec(in_names);   \
+    const std::vector<std::string> out_names_vec(out_names); \
+    GEN_VAR_COMMON(__VA_ARGS__)
+// make common except paddle OpExtension
+#define MAKE_MAP_COMMON(FRAMEWORK, ...)                                                          \
+    GEN_VAR_COMMON(__VA_ARGS__)                                                                  \
+    if (!name.empty())                                                                           \
+        return std::make_shared<ov::frontend::FRAMEWORK::OpExtension<T>>(name, attr_mp, val_mp); \
+    return std::make_shared<ov::frontend::FRAMEWORK::OpExtension<T>>(attr_mp, val_mp);
+#define MAKE_MAP_onnx(...)       MAKE_MAP_COMMON(onnx, __VA_ARGS__)
+#define MAKE_MAP_tensorflow(...) MAKE_MAP_COMMON(tensorflow, __VA_ARGS__)
+// make paddle OpExtension
+#define MAKE_MAP_paddle(...)                                                         \
```

Comment:
Why do we have paddle specific macros in common header?

Reply:
In order to make the original 'OPENVINO_FRAMEWORK_MAP' work for paddle, and satisfy the paddle needs input tensor names and output tensor names, the macro for paddle should be: 
`OPENVINO_FRAMEWORK_MAP(paddle, {"input name"}, {"output name"}, ...)`
Because the macro parameters list are different between paddle and other frontends, here does some trick to extract specific parameters for paddle.

Anyway the solution is not good, there is some difference between paddle and other frontends:
1, paddle ops must use input tensor names and output tensor names as mentioned.
2, NodeContext lacks iteration interface to enumerate the input/output tensor names.
So seems could not use the same approach as onnx/tf.